<a href="https://colab.research.google.com/github/kumaramardeep342/Colab-Work/blob/main/Ed_AI__NLP_M2%2BM3_inclass_ATIS_Intent_classification_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Summary - How to predict model on NLP Data
0. Load the dataset
1. Import libraries
2. Read the dataset
3. Data Preprocessing on X_data and Y_data
4. Split the dataset into X_data and Y_data
5. Split the data into training and testing
6. Use only training data to train the model
7. Hyperparameter tunning
8. pass the test to the trained model to predict y_pred_test - validation metrics
9. Create a samml API - x_data[response] - post - "airfare" etc.

# Loading the Dataset

In [35]:
! pip install -q opendatasets
import opendatasets as od
od.download('https://www.kaggle.com/datasets/hassanamin/atis-airlinetravelinformationsystem')

Skipping, found downloaded files in "./atis-airlinetravelinformationsystem" (use force=True to force download)


# About The Dataset & Model
The ATIS dataset is a standard benchmark dataset widely used as an intent classification. ATIS Stands for Airline Travel Information System. Intent classification is an important component of Natural Language Understanding (NLU) systems in any chatbot platform.

Within a chatbot, intent refers to the goal the customer has in mind when typing in a question or comment. While entity refers to the modifier the customer uses to describe their issue, the intent is what they really mean. For example, a user says, ‘I need new shoes.’ The intent behind the message is to browse the footwear on offer.


# 1.Import libraries

In [36]:
# 1. Import libraries
import pandas as pd  #for redaing  .csv dataset

#enable table format
from google.colab import data_table
data_table.enable_dataframe_formatter()
#disable table format
# from google.colab import data_table
# data_table.disable_dataframe_formatter()


# 2.Read the dataset

In [37]:
df=pd.read_csv('/content/atis-airlinetravelinformationsystem/atis_intents.csv')
df.columns = ['Intent','Text'] #Re-naming the columns
df.head()

Intent                                               Text
0       atis_flight   what flights are available from pittsburgh to...
1  atis_flight_time   what is the arrival time in san francisco for...
2      atis_airfare            cheapest airfare from tacoma to orlando
3      atis_airfare   round trip fares from pittsburgh to philadelp...
4       atis_flight   i need a flight tomorrow from columbus to min...

In [38]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4977 entries, 0 to 4976
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Intent  4977 non-null   object
 1   Text    4977 non-null   object
dtypes: object(2)
memory usage: 77.9+ KB


In [39]:
df.Intent.value_counts()
#len(df.Intent.value_counts())

Intent
atis_flight                                 3665
atis_airfare                                 423
atis_ground_service                          255
atis_airline                                 157
atis_abbreviation                            147
atis_aircraft                                 81
atis_flight_time                              54
atis_quantity                                 51
atis_flight#atis_airfare                      21
atis_airport                                  20
atis_distance                                 20
atis_city                                     19
atis_ground_fare                              18
atis_capacity                                 16
atis_flight_no                                12
atis_meal                                      6
atis_restriction                               6
atis_airline#atis_flight_no                    2
atis_ground_service#atis_ground_fare           1
atis_airfare#atis_flight_time                  1
atis_cheapest

>Findings
- There are 4977 entries and 22 different type of intents

In [40]:
# get rid of some columns where intent counts is less than 10
#atis_flight#atis_airfare
# atis_meal
# atis_restriction
# atis_airline#atis_flight_no
# atis_ground_service#atis_ground_fare
# atis_airfare#atis_flight_time
# atis_cheapest
# atis_aircraft#atis_flight#atis_flight_no
